serverless/cmd/feeder/github/main.go

```diff
@@ -0,0 +1,561 @@
+	}
+
+	glog.V(1).Infof("CP after feeding:\n%s", string(wRaw))
+
+	branchName := fmt.Sprintf("refs/heads/witness_%s", base64.StdEncoding.EncodeToString(wCp.Hash))
+	deleteBranch, err := gitCreateLocalBranch(forkRepo, headRef, branchName)
+	if err != nil {
+		return fmt.Errorf("failed to create git branch for PR: %v", err)
+	}
+	defer deleteBranch()
+
+	outputPath := filepath.Join(opts.logPath, "checkpoint.witnessed")
+	msg := fmt.Sprintf("Witness checkpoint@%v", wCp.Size)
+	if err := gitCommitFile(workTree, outputPath, wRaw, msg, opts.gitUsername, opts.gitEmail); err != nil {
+		return fmt.Errorf("failed to commit updated checkpoint.witnessed file: %v", err)
+	}
+
+	// 5. git force-push to origin/branchrefname
+	glog.V(1).Infof("git push -f origin/%s", branchName)
+	if err := forkRepo.Push(&git.PushOptions{
+		Force: true,
+	}); err != nil {
+		return fmt.Errorf("git push -f: %v", err)
+	}
+
+	// 6. Create GH pull request
+	// 7. git checkout master  (done in defer)
+	// 8. delete branch branchRefName  (done in defer)
+	return createPR(ctx, opts, ghCli, fmt.Sprintf("Witness @ %d", wCp.Size), opts.witnessRepo.owner+":"+branchName, "master")
+}
+
+// gitCreateLocalBranch creates a local branch on the given repo.
+//
+// Returns a function which will delete the local branch when called.
+func gitCreateLocalBranch(repo *git.Repository, headRef *plumbing.Reference, branchName string) (func(), error) {
+	// Create a git branch for the witnessed checkpoint to be added, named
+	// using the base64(checkpoint hash).  Construct a fully-specified
+	// reference name for the new branch.
+	branchRefName := plumbing.ReferenceName(branchName)
+	branchHashRef := plumbing.NewHashReference(branchRefName, headRef.Hash())
+
+	workTree, err := repo.Worktree()
+	if err != nil {
+		return nil, fmt.Errorf("failed to get work tree: %v", err)
+	}
+
+	// git checkout `branchRefName`
+	glog.V(1).Infof("git checkout -b %q", branchHashRef.Name())
+	if err := workTree.Checkout(&git.CheckoutOptions{
+		Branch: branchRefName,
+		Create: true,
+	}); err != nil {
+		return nil, fmt.Errorf("git checkout %v: %v", branchRefName, err)
+	}
+	d := func() {
+		if err := repo.Storer.RemoveReference(branchHashRef.Name()); err != nil {
+			glog.Errorf("failed to delete branch hashref for %q: %v", branchHashRef.Name(), err)
+		}
+		glog.V(1).Infof("git branch -D %v", branchHashRef.Name())
+		if err := workTree.Checkout(&git.CheckoutOptions{
+			Branch: "refs/heads/master",
+		}); err != nil {
+			glog.Errorf("failed to git checkout master: %v", err)
+			return
+		}
+		glog.V(1).Info("git checkout master")
+	}
+
+	return d, nil
+}
+
+// gitCommitFile creates a commit on a repo's worktree which overwrites the specifed file path
+// with the provided bytes.
+func gitCommitFile(workTree *git.Worktree, path string, raw []byte, commitMsg string, username, email string) error {
+	// 2. serialize witnessed CP to file
+	glog.Infof("Writing %q", path)
```

Comment:
```suggestion
	glog.Infof("Writing checkpoint (%d bytes) to %q", len(raw), path)
```

Suggestion for reducing the comments and info message to one thing that's more useful.

Reply:
done